File: 438. Find All Anagrams in a String/slidewindow.py
class Solution:
    def findAnagrams(self, s: str, p: str):
        n, l = len(s), len(p)
        res = []
        vs, vp = [0] * 26, [0] * 26
        
        for c in p:
            vp[ord(c) - ord('a')] += 1
            
        for i in range(n):
            if i >= l:
                vs[ord(s[i - l]) - ord('a')] -= 1
            vs[ord(s[i]) - ord('a')] += 1
            if (vs == vp):
                res.append(i + 1 - l)
        return res

File: 438. Find All Anagrams in a String/slidewindow.cpp
class Solution {
public:
    vector<int> findAnagrams(string s, string p) {
            int n = s.length(), l = p.length();
            vector<int> res;
            vector<int> vp(26, 0), vs(26, 0);

            for (char c: p)
                ++vp[c - 'a'];

            for (int i = 0; i < n; i++) {
                if (i >= l)
                    --vs[s[i - l] - 'a'];
                ++vs[s[i] - 'a'];
            if (vs == vp)
                res.push_back(i + 1 - l);
            }
            return res;
        }
};